app/src/main/java/in/ac/bits_pilani/goa/ard/activities/AboutMACActivity.java

```diff
@@ -0,0 +1,135 @@
+package in.ac.bits_pilani.goa.ard.activities;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.Toolbar;
+import android.util.Log;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import com.bumptech.glide.Glide;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.database.ValueEventListener;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import in.ac.bits_pilani.goa.ard.R;
+import in.ac.bits_pilani.goa.ard.utils.AHC;
+
+import static in.ac.bits_pilani.goa.ard.utils.AHC.TIAB1;
+import static in.ac.bits_pilani.goa.ard.utils.AHC.TIAB2;
+import static in.ac.bits_pilani.goa.ard.utils.AHC.TIAB3;
+import static in.ac.bits_pilani.goa.ard.utils.AHC.TIAB4;
+import static in.ac.bits_pilani.goa.ard.utils.AHC.TIAB5;
+import static in.ac.bits_pilani.goa.ard.utils.AHC.TIAB6;
+
+/**
+ * AboutMac class.
+ * @author Aayush
+ * @version 1.0.
+ */
+
+public class AboutMACActivity extends AppCompatActivity {
+
+    /**
+     * About Mac Text.
+     */
+    @BindView(R.id.about_mac_text)
+    TextView textView;
+
+    /**
+     * About Mac Image.
+     */
+    @BindView(R.id.about_mac_image)
+    ImageView imageView;
+
+    /**
+     * About Mac Toolbar.
+     */
+    @BindView(R.id.toolbar_about_mac)
+    Toolbar toolbar;
+
+    /**
+     * Database Reference.
+     */
+    private DatabaseReference databaseReference;
+
+    /**
+     * Vale event listener.
+     */
+    private ValueEventListener listener;
+
+    /**
+     * Sharedpreferences.
+     */
+    private SharedPreferences sharedPreferences;
+
+    /**
+     * String text.
+     */
+    private String text;
+
+    /**
+     * String url.
+     */
+    private String url;
+
+    /**
+     * Tag for about mac activity.
+     */
+    private final String TAG = AHC.TAG + ".activities." + getClass().getSimpleName();
+
+    @Override
+    protected void onCreate(final Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_about_mac);
+        ButterKnife.bind(this);
+        setSupportActionBar(toolbar);
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+    }
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+        sharedPreferences = getSharedPreferences(AHC.USER_PREFERENCES, Context.MODE_PRIVATE);
+        text = sharedPreferences.getString(AHC.TEXT_ABOUT_MAC, "");
+        url = sharedPreferences.getString(AHC.IMAGEURL_ABOUT_MAC, "");
+        if ("".equals(text)) {
+            textView.setText(TIAB1 + TIAB2 + TIAB3 + TIAB4 + TIAB5 + TIAB6);
```

Comment:
Try R.string and for coverage use Mockito to mock the sharedpreferences file

If at the end it still doesn't work out we can add a suppression in checkstyle (But I think it is possible by method 1)